### DISCUSSION TITLE
Does brew upgrade include brew update?

### DISCUSSION THREAD

**Discussion by johnmapeson** [Homebrew/discussions#5278] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0476c2e5e4f02b409084553896340b2f4ab3352e
Last commit: 5 days ago
Core tap JSON: 08 Apr 09:24 UTC
Core cask tap JSON: 08 Apr 09:24 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

[I was told](https://github.com/orgs/Homebrew/discussions/5146#discussioncomment-8553202) that `brew update` is already run by `brew upgrade` in modern Homebrew, so running `brew update` before `brew upgrade` is redundant.

Is it really so? I'm asking this because I don't see it mentioned anywhere else. For example, here: https://mac.install.guide/homebrew/4

**Comment by SMillerDev** [comment, marked as answer]
Yes it is, unless you explicitly disable it. 

In general members of the Homebrew organisation are a pretty reliable source of Homebrew actions compared to random guides online. This guide is wrong on various other aspects of Homebrew behaviour too.

**Reply by johnmapeson** [reply]
Hi, Sean!

This is true not only for `brew upgrade`, but also for `brew upgrade <formula>`, right?

And out of curiosity, what might be a case to prefer  `brew update` on its own, and not as a part of `brew upgrade`?

**Reply by SMillerDev** [reply]
> This is true not only for `brew upgrade`, but also for `brew upgrade <formula>`, right?

Yes
> And out of curiosity, what might be a case to prefer `brew update` on its own, and not as a part of `brew upgrade`?

If you don't want things to upgrade.
